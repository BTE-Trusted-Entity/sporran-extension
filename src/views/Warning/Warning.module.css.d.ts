export const __esModule: true;
export const container: string;
export const heading: string;
export const important: string;
export const info: string;
export const confirm: string;
export const cancel: string;
