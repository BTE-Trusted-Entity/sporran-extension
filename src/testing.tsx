export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
