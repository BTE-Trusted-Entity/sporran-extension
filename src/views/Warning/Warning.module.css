.container {
  composes: container from '../../components/View/View.module.css';
}

.heading {
  composes: screenHeadline from '../../components/Typography/Typography.module.css';
  color: var(--color-orange);
  background-image: url('../../components/images/attention-hand.svg');
  background-size: calc(86rem / 16) calc(118rem / 16);
  background-repeat: no-repeat;
  background-position: center 3.25rem;
  padding-bottom: calc(156rem / 16);
}

.important {
  composes: importantHeadline from '../../components/Typography/Typography.module.css';
  margin: 0;
  white-space: pre-line;
}

.info {
  composes: subline from '../../components/Typography/Typography.module.css';
  margin: 0.5rem;
}

.confirm {
  composes: buttonPrimary from '../../components/Button/Button.module.css';
  margin-top: 1.5rem;
}

.cancel {
  composes: buttonSecondary from '../../components/Button/Button.module.css';
  margin-top: 1rem;
}
