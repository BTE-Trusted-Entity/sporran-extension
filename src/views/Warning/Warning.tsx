import { browser } from 'webextension-polyfill-ts';
import { Link } from 'react-router-dom';

import * as styles from './Warning.module.css';

import { LinkBack } from '../../components/LinkBack/LinkBack';
import { paths } from '../paths';

export function Warning(): JSX.Element {
  const t = browser.i18n.getMessage;

  return (
    <div className={styles.container}>
      <h1 className={styles.heading}>{t('view_Warning_headline')}</h1>
      <h2 className={styles.important}>{t('view_Warning_emphasis')}</h2>
      <p className={styles.info}>{t('view_Warning_explanation')}</p>
      <h2 className={styles.important}>{t('view_Warning_emphasis_again')}</h2>
      <Link to={paths.identity.create.backup} className={styles.confirm}>
        {t('view_Warning_CTA')}
      </Link>
      <Link to={paths.home} className={styles.cancel}>
        {t('common_action_cancel')}
      </Link>

      <LinkBack />
    </div>
  );
}
