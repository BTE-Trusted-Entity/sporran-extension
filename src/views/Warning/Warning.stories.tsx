import { Meta } from '@storybook/react';

import { Warning } from './Warning';

export default {
  title: 'Views/Warning',
  component: Warning,
} as Meta;

export function Template(): JSX.Element {
  return <Warning />;
}
