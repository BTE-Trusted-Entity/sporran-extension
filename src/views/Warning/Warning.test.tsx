import { render } from '../../testing/testing';

import { Warning } from './Warning';

describe('Warning', () => {
  it('should render', () => {
    const { container } = render(<Warning />);
    expect(container).toMatchSnapshot();
  });
});
