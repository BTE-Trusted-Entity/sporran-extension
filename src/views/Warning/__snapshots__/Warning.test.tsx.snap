// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warning should render 1`] = `
<div>
  <div
    class="container"
  >
    <h1
      class="heading"
    >
      Attention!
    </h1>
    <h2
      class="important"
    >
      The next step is very important!
    </h2>
    <p
      class="info"
    >
      Please write down your wallet’s phrase and keep it in a safe place. We recommend that you write down your backup phrase with pen and paper. The phrase is required to restore your Identity.
    </p>
    <h2
      class="important"
    >
      Keep it safe, so you don’t lose
your assets
    </h2>
    <a
      class="confirm"
      href="/identity/create/backup"
    >
      I understand this – go to the phrase
    </a>
    <a
      class="cancel"
      href="/"
    >
      Cancel
    </a>
    <button
      aria-label="Back"
      class="linkBack"
      title="Back"
      type="button"
    />
  </div>
</div>
`;
